File: VB/CustomizeReportDataExample.Module.Win/CustomReportWizardParameters.vb
Imports System.Linq
Imports System.Text
Imports DevExpress.ExpressApp.DC
Imports DevExpress.ExpressApp.ReportsV2
Imports DevExpress.ExpressApp.ReportsV2.Win
Imports DevExpress.XtraReports.UI
Imports CustomizeReportDataExample.Module.BusinessObjects


Namespace CustomizeReportDataExample.Module.Win
    <DomainComponent> _
    Public Class CustomReportWizardParameters
        Inherits NewReportWizardParameters

        Public Sub New(ByVal report As XtraReport, ByVal reportDataType As Type)
            MyBase.New(report, reportDataType)
        End Sub
        Public Property Category() As String
        Public Overrides Sub AssignData(ByVal reportData As IReportDataV2Writable)
            MyBase.AssignData(reportData)
            If TypeOf reportData Is MyReportDataV2 Then
                CType(reportData, MyReportDataV2).Category = Category
            End If
        End Sub
    End Class
End Namespace

File: CS/CustomizeReportDataExample.Module.Win/CustomReportWizardParameters.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.ExpressApp.DC;
using DevExpress.ExpressApp.ReportsV2;
using DevExpress.ExpressApp.ReportsV2.Win;
using DevExpress.XtraReports.UI;
using CustomizeReportDataExample.Module.BusinessObjects;


namespace CustomizeReportDataExample.Module.Win {
    [DomainComponent]
    public class CustomReportWizardParameters : NewReportWizardParameters {
        public CustomReportWizardParameters(XtraReport report, Type reportDataType) :
            base(report, reportDataType) { }
        public string Category { get; set; }
        public override void AssignData(IReportDataV2Writable reportData) {
            base.AssignData(reportData);
            if (reportData is MyReportDataV2) {
                ((MyReportDataV2)reportData).Category = Category;
            }
        }
    }
}
